### DISCUSSION TITLE
Workflow dynamic configuration for transition

### DISCUSSION THREAD

**Discussion by lpiccini21** [pimcore/pimcore#7772] [General]
Hi guys,
is there a way to add a select to additionalFields of transition workflow (for an object) with options from optionsProvider class that could return different options for each object?

And is it possible to send the notification email to user dynamically, depending on user assigned to the object (it is a field of the object).

Thanks

**Comment by neha04** [comment]
@lpiccini21 were you able to add a select to additionalFields of transition workflow (for an object) with options from optionsProvider class?

**Reply by neha04** [reply]
@dvesh3 do you think its possible in current pimcore version?

**Reply by lpiccini21** [reply]
Hi Neha,
I didn't find a way to use optionsProvider directly on config.yml file.
I use a listener on event pimcore.admin.dataobject.get.preSendData with negative priority to intercept the data sent to the controller that display the object page.
The listener method get the argument "data" that contain the workflow configuration for the current object and manipulate the AdditionalFields dinamically.

The priority is negative because the same listener is used by Pimcore Bundle to show the workflow transition buttons and my listener must be executed after that.

config.yml

`AppBundle\EventListener\MyObjectListener:`
`     tags:`
`            - { name: kernel.event_listener, event: pimcore.admin.dataobject.get.preSendData, method: onPreSendData, priority: -50 }`
